cg/services/illumina_services/illumina_post_processing_service/illumina_post_processing_service.py

```diff
@@ -115,10 +116,32 @@ def store_sequencing_data_in_status_db(
         sequencing_run: IlluminaSequencingRun = self.store_illumina_sequencing_run(
             run_directory_data=run_directory_data, flow_cell=flow_cell
         )
-        self.store_illumina_sample_sequencing_metrics(
-            run_directory_data=run_directory_data, sequencing_run=sequencing_run
+        sample_metrics: list[IlluminaSampleSequencingMetricsDTO] = (
+            self.store_illumina_sample_sequencing_metrics(
+                run_directory_data=run_directory_data, sequencing_run=sequencing_run
+            )
         )
         self.status_db.commit_to_store()
+        self.update_samples_for_metrics(
```

Comment:
Does it make sense to do the commit_to store before updating the sample metrics? Can skip the commit in `update_sample_reads_illumina` and `update_sample_sequenced_at`, and instead jus move this line?

Reply:
There is a commit in the update function. Do you want to move that out of there then?

Reply:
Since we use the same session it should be fine. `samples.sample_run_metrics` should give you all the metrics even though you haven't run commit.

Reply:
Ok, then we should revisit how we write the CRUD update, delete and create functions. Right now, the update, create and delete all commit within them. I will create an issue to track this and that we potentially want to change that behaviour. But we should be consistent with it.